pkg/apis/core/validation/shoot.go

```diff
@@ -2742,6 +2767,80 @@ func ValidateForceDeletion(newShoot, oldShoot *core.Shoot) field.ErrorList {
 	return allErrs
 }
 
+// ValidateInPlaceUpdates validates the updates of the worker pools with in-place update strategy of a Shoot.
+func ValidateInPlaceUpdates(newShoot, oldShoot *core.Shoot) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Allow the update without validation if the force-update annotation is present.
+	if helper.ShootNeedsForceInPlaceUpdate(newShoot) {
+		return allErrs
+	}
+
+	if newShoot.Status.InPlaceUpdates == nil || newShoot.Status.InPlaceUpdates.PendingWorkersRollouts == nil {
+		return allErrs
+	}
+
+	pendingWorkerNames := sets.New[string]()
+
+	pendingWorkerNames.Insert(newShoot.Status.InPlaceUpdates.PendingWorkersRollouts.AutoInPlaceUpdate...)
+	pendingWorkerNames.Insert(newShoot.Status.InPlaceUpdates.PendingWorkersRollouts.ManualInPlaceUpdate...)
+
+	for i, worker := range newShoot.Spec.Provider.Workers {
+		if !helper.IsUpdateStrategyInPlace(worker.UpdateStrategy) {
+			continue
+		}
+
+		if !pendingWorkerNames.Has(worker.Name) {
+			continue
+		}
+
+		var (
+			oldWorker core.Worker
+			idxPath   = field.NewPath("spec", "provider", "workers").Index(i)
+		)
+
+		oldWorkerIndex := slices.IndexFunc(oldShoot.Spec.Provider.Workers, func(ow core.Worker) bool {
+			oldWorker = ow
+			return ow.Name == worker.Name
+		})
+
+		if oldWorkerIndex == -1 {
+			continue
+		}
+
+		oldWorkerIndexPath := field.NewPath("spec", "provider", "workers").Index(oldWorkerIndex)
+		oldControlPlaneVersion, err := semver.NewVersion(oldShoot.Spec.Kubernetes.Version)
+		if err != nil {
+			allErrs = append(allErrs, field.Forbidden(oldWorkerIndexPath, "old control plane version is not a valid semver version"))
```

Comment:
same for below lines 
```suggestion
			allErrs = append(allErrs, field.Forbidden(oldWorkerIndexPath, "old control plane kubernetes version is not a valid semver version"))
```